src/core/thread.d

```diff
@@ -631,7 +633,8 @@ class Thread
         }
 
         // Start thread as non-suspendable, undone in thread_entryPoint
-        atomicStore(m_isInCriticalRegion, true);
+        synchronized (Thread.criticalRegionLock)
+            m_isInCriticalRegion = true;
```

Comment:
> So thread_suspendAll wouldn't work when being called from DllMain.

It actually will work, the newly started will leave the critical section right after calling Thread.setThis.
The only thing we need to take care of, is to not acquire the loader lock while in the critical section.